fdbrpc/libcoroutine/Coro.c

```diff
@@ -66,6 +66,8 @@ VALGRIND_STACK_DEREGISTER((coro)->valgrindStackId)
 #define STACK_DEREGISTER(coro)
 #endif
 
+#pragma GCC diagnostic ignored "-Wreturn-local-addr"
```

Comment:
This will fail on clang (Mac OS X)

    fdbrpc/libcoroutine/Coro.c:175:32: error: unknown warning group '-Wreturn-local-addr', ignored [-Werror,-Wunknown-warning-option]

I suggest wrapping it into #if defined(...):

    #if defined(__GNUC__) && !defined(__clang__)
    #pragma GCC diagnostic ignored "-Wreturn-local-addr"
    #endif

Reply:
Here is mine:

    Antons-MacBook-Pro:foundationdb anton$ gcc --version
    Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
    Apple LLVM version 9.1.0 (clang-902.0.39.1)
    Target: x86_64-apple-darwin17.5.0
    Thread model: posix
    InstalledDir: /Library/Developer/CommandLineTools/usr/bin